src/borrower/feed/navfeedPV.sol

```diff
@@ -0,0 +1,222 @@
+        lastNAVUpdate = uniqueDayTimestamp(block.timestamp);
+        emit Rely(msg.sender);
+    }
+
+    function toUint128(uint256 value) internal pure returns (uint128) {
+        require(value <= type(uint128).max, "SafeCast: value doesn't fit in 128 bits");
+        return uint128(value);
+    }
+    
+    // returns the ceiling of a loan
+    // the ceiling defines the maximum amount which can be borrowed
+    function ceiling(uint loan) public virtual view returns (uint) {
+        bytes32 nftID_ = nftID(loan);
+        uint initialCeiling = rmul(nftValues(nftID_), ceilingRatio(risk(nftID_)));
+        return safeSub(initialCeiling, pile.debt(loan));
+    }
+
+    // --- Administration ---
+    function depend(bytes32 contractName, address addr) external auth {
+        if (contractName == "pile") {pile = PileLike(addr);}
+        else if (contractName == "shelf") { shelf = ShelfLike(addr); }
+        else revert();
+        emit Depend(contractName, addr);
+    }
+
+    function file(bytes32 name, uint risk_, uint thresholdRatio_, uint ceilingRatio_, uint rate_) public auth {
+        if(name == "riskGroup") {
+            require(ceilingRatio(risk_) == 0, "risk-group-in-usage");
+            riskGroup[risk_].thresholdRatio = toUint128(thresholdRatio_);
+            riskGroup[risk_].ceilingRatio = toUint128(ceilingRatio_);
+
+            // set interestRate for risk group
+            pile.file("rate", risk_, rate_);
+            emit File(name, risk_, thresholdRatio_, ceilingRatio_, rate_);
+
+        } else { revert ("unknown name");}
+    }
+
+    // --- Actions ---
+    function borrow(uint loan, uint amount) external virtual auth returns(uint navIncrease) {
+        require(ceiling(loan) >= amount, "borrow-amount-too-high");
+        return amount;
+    }
+
+    function repay(uint loan, uint amount) external virtual auth {
+    }
+
+    function borrowEvent(uint loan, uint) public virtual auth {
+        uint risk_ = risk(nftID(loan));
+        // when issued every loan has per default interest rate of risk group 0.
+        // correct interest rate has to be set on first borrow event
+        if (pile.loanRates(loan) != risk_) {
+            // set loan interest rate to the one of the correct risk group
+            pile.setRate(loan, risk_);
+        }
+    }
+
+    function repayEvent(uint loan, uint amount) public virtual auth {}
+    function lockEvent(uint loan) public virtual auth {}
+    function unlockEvent(uint loan) public virtual auth {}
+
+    function writeOff(uint loan) public auth {
+       pile.changeRate(loan, WRITEOFF_RATE_GROUP);
+    }
+
+    function isLoanWrittenOff(uint loan) public view returns(bool) {
+        return pile.loanRates(loan) == WRITEOFF_RATE_GROUP;
+    }
+
+    // --- NAV calculation ---
+    function currentNAV() public view returns(uint) {
+        uint totalDebt;
+        // calculate total debt
+        for (uint loanId = 1; loanId <= shelf.loanCount(); loanId++) {
```

Comment:
```suggestion
        for (uint loanId = 1; loanId < shelf.loanCount(); loanId++) {
```
If you iterate over the loans. I think it should just be `<`. 

```solidity
    function loanCount() public view returns (uint256 totalNumber) {
        return title.count();
    }
```

**Example**.

Here is an example to illustrate.
`Title.count` starts with 1. (zero loans in the system.)

After the first loan (id = 1) `count` will be `2`.

I think this is actually a weird implementation. I don't remember why Lucas back then choose to start with loan id =1.